Document/0x04f-Testing-Network-Communication.md

```diff
@@ -108,14 +108,150 @@ If that's the case, you are now able to see the complete network traffic that is
+
+Simple way to simulate a man-in-the-middle (MITM) attack is to configure network where all packets between devices and the target network are going through your machine. In mobile world, this is achieved by placing Access Point between mobile devices and your machine. Your machine is a router and an Access Point.
+
+Following scenarios are possible:
+- use your machine built-in WiFi card as an Access Point and use your wired connection to connect to the target network,
+- use external USB WiFi card as an Access Point and user your machine built-in WiFi to connect to the target network (can be vice-versa) or
+- use external Access Point and redirect traffic to your machine.
+
+Scenarios with WiFi card Access Point require that WiFi card has AP (Access Point) capabilities. Additionally, you need to install tools and/or configure network to reflect man-in-the-middle position (see below).
+Scenario with external Access Point requires access to the configuration of the AP and you should check first if the AP supports either:
 
 - port forwarding or
 - has a span or mirror port.
 
-In both scenarios the AP needs to be configured to point to your machines IP. Tools like Wireshark can then again be used to monitor and record the traffic for further investigation.
+In both cases the AP needs to be configured to point to your machines IP. Your machine must be connected to the AP (via wired connection or Wifi) and you need to have connection to the target network (can be by the same connection as to the AP). Some additional configuration may be required on your machine to route traffic to the target network.
+
+> If external Access Point belongs to customer, all changes and configurations should be clarified prior to the engagement.
+
+<img src="Images/Chapters/0x04f/architecture_MITM_AP.png" alt="Network Diagram - MITM with an Access Point">
+
+##### Installation
+
+The following procedure is setting up a man-in-the-middle position using an Access Point and additional network interface:
+1.  Create an Access Point and its network.
+2.  Route traffic to additional network interface where traffic can reach the target network. Additional network interface can be wired connection or other Wifi card.
+
+This can be done by using the built-in utilities on macOS. You can use [share the internet connection on Mac with other network users](https://support.apple.com/en-ke/guide/mac-help/mchlp1540/mac "Share the internet connection on Mac with other network users").
+
+For all major Linux and Unix operating systems you need:
+- hostapd,
+- dnsmasq,
+- iptables,
+- wpa_supplicant,
+- airmon-ng or
+- its equivalents.
+
+For Kali Linux you can install these tools with `apt-get`:
+
+```shell
+$ apt-get update
+$ apt-get install hostapd dnsmasq aircrack-ng
+```
+> iptables and wpa_supplicant are installed by default on Kali Linux.
+
+You should check if your one of your Wifi card has AP capabilities. This can be done by using the command `iwconfig` on Kali Linux:
```

Comment:
This can be part of line 118, "Use your machines built-in WiFi card as an Access Point and use your wired connection to connect to the target network. You can verify if your Wifi card has AP capabilites..."